jib-core/src/main/java/com/google/cloud/tools/jib/builder/RetrieveRegistryCredentialsStep.java

```diff
@@ -65,6 +66,24 @@ public Authorization call() throws IOException, NonexistentDockerCredentialHelpe
         return buildConfiguration.getKnownRegistryCredentials().getAuthorization(registry);
       }
 
+      // Tries to infer common credential helpers for known registries.
+      for (String registrySuffix : COMMON_CREDENTIAL_HELPERS.keySet()) {
+        if (registry.endsWith(registrySuffix)) {
+          try {
+            Authorization authorization =
+                retrieveFromCredentialHelper(COMMON_CREDENTIAL_HELPERS.get(registrySuffix));
```

Comment:
This method (`retreiveFromCredentialHelper`) seems to be catching the NonexistantDockerCredentialHelperException on line 110 below?

Reply:
Yea, since we are inferring, the build should not break if the inferred credential helper was not found. For example, the user could be pulling a public image on gcr.io that does not require credentials.

Reply:
oops, okay I read the names wrong.